frontend/src/components/AssignmentDialog/AssignmentInfoCard.vue

```diff
@@ -0,0 +1,249 @@
+
+        <v-btn
+          v-if="userState.role == ROLES.insurance" 
+          class="mt-2"
+          color="info"
+          variant="elevated"
+          @click="openCreateCommentModal"
+        >
+          + Add comment
+        </v-btn>
+      </v-window-item>
+    </v-window>
+  </v-col>
+
+  <CreateCommentModal 
+    v-if="isCreateCommentModalOpen"
+    v-model="isCreateCommentModalOpen"
+    @close-modal="closeCreateCommentModal"
+  />
+
+  <CommentModal
+    v-if="isCommentModalOpen"
+    v-model="isCommentModalOpen"
+    :comment-data="selectedCommentData"
+    @close-modal="closeCommentModal"
+  />
+
+  <SetRepairFacilityModal 
+    v-if="isSetRepairFacilityModalOpen"
+    v-model="isSetRepairFacilityModalOpen"
+    @close-modal="closeSetRepairFacilityModal"
+  />
+</template>
+
+<script setup lang="ts">
+  import { onMounted, onUnmounted, ref } from 'vue';
+  import { useEventBus } from '@vueuse/core';
+  import { storeToRefs } from 'pinia';
+
+  import { eventBusNames, ROLES } from '@/helpers/constants';
+  import { formatDate, formatDateWithTime } from '@/helpers/assignment';
+  import { getChipStatusColor, getChipStatusIcon } from '@/helpers/styles';
+  import { useUserStore } from '@/stores/user';
+  import { useAssignmentStore } from '@/stores/assignment';
+  import { useGetComments } from '@/hooks/useGetComments';
+
+  import CreateCommentModal from '@/components/CreateCommentModal.vue';
+  import CommentModal from '@/components/CommentModal.vue';
+  import SetRepairFacilityModal from '@/components/SetRepairFacilityModal.vue';
+
+  import type { IComment } from '@/helpers/interfaces';
+
+  const { userState } = storeToRefs(useUserStore());
+  const { assignment } = storeToRefs(useAssignmentStore());
+
+  const bus = useEventBus<string>(eventBusNames.fetchComments);
+  const tab = ref(null);
+  const isCreateCommentModalOpen = ref(false);
+  const isSetRepairFacilityModalOpen = ref(false);
+  const isCommentModalOpen = ref(false);
+  const selectedCommentData = ref({});
+
+  const { getComments } = useGetComments();
+
+  const getCommentItemClasses = (isCommentRead: boolean) => {
+    let classes = 'comment-item';
+
+    if (userState.value.role == ROLES.repair && !isCommentRead) {
+      classes += ' comment-item-unread';
+    }
+
+    return classes;
+  };
+  
+  const openCreateCommentModal = () => {
+    isCreateCommentModalOpen.value = true;
+  };
+
+  const closeCreateCommentModal = () => {
+    isCreateCommentModalOpen.value = false;
+  };
+
+  const openCommentModal = (data: IComment) => {
+    selectedCommentData.value = data;
+    isCommentModalOpen.value = true;
+  };
+
+  const closeCommentModal = () => {
+    isCommentModalOpen.value = false;
+  };
+
+  const openSetRepairFacilityModal = () => {
+    isSetRepairFacilityModalOpen.value = true;
+  };
+
+  const closeSetRepairFacilityModal = () => {
+    isSetRepairFacilityModalOpen.value = false;
+  };
+
+  const handleBusComments = (busEvent: string) => {

```

Comment:
why string and not boolean? String checks are much slower than boolean

Reply:
Done